logstash-core/api/lib/app/stats.rb

```diff
@@ -17,7 +31,25 @@ class Stats < BaseApp
     # - events in the pipeline
     get "/events" do
       command = factory.build(:events_command)
-      respond_with command.run
+      respond_with({ :events => command.run })
+    end
+
+    # return hot threads information
+    get "/jvm/hot_threads" do
+      top_threads_count = params["threads"].to_i || 3
+      ignore_idle_threads = params["ignore_idle_threads"] || true
+      options = {
+        :threads => top_threads_count,
+        :ignore_idle_threads => as_boolean(ignore_idle_threads)
+      }
```

Comment:
We can do it, what about doing it in a feature PR?

- purbon

On Mon, Jan 18, 2016 at 6:02 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In logstash-core/api/lib/app/stats.rb
> <https://github.com/elastic/logstash/pull/4472#discussion_r50022008>:
>
> > @@ -17,7 +31,25 @@ class Stats < BaseApp
> >      # - events in the pipeline
> >      get "/events" do
> >        command = factory.build(:events_command)
> > -      respond_with command.run
> > +      respond_with({ :events => command.run })
> > +    end
> > +
> > +    # return hot threads information
> > +    get "/jvm/hot_threads" do
> > +      top_threads_count = params["threads"].to_i || 3
> > +      ignore_idle_threads = params["ignore_idle_threads"] || true
> > +      options = {
> > +        :threads => top_threads_count,
> > +        :ignore_idle_threads => as_boolean(ignore_idle_threads)
> > +      }
>
> Did you thought about using an class for that and we could just pass if to
> the command.run.
> This class could take care of the default values and values casting?
>
> We can make a class for each endpoint that requires parsing user data.
>
> IIRC in django/rails world we call them form object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4472/files#r50022008>.
>